extension/extensionlimiter/limiterhelper/wrapper.go

```diff
@@ -0,0 +1,125 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package limiterhelper // import "go.opentelemetry.io/collector/extension/extensionlimiter/limiterhelper"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/extension/extensionlimiter"
+)
+
+// LimiterWrapperProvider provides access to LimiterWrappers, which is
+// the appropriate interface for callers that can easily wrap a
+// function call, because for wrapped calls there is no distinction
+// between rate limiters and resource limiters.
+type LimiterWrapperProvider interface {
+	extensionlimiter.CheckerProvider
+
+	GetLimiterWrapper(extensionlimiter.WeightKey, ...extensionlimiter.Option) (LimiterWrapper, error)
+}
+
+// GetLimiterWrapperFunc is an easy way to build GetLimiterWrapper functions.
+type GetLimiterWrapperFunc func(extensionlimiter.WeightKey, ...extensionlimiter.Option) (LimiterWrapper, error)
+
+// GetLimiterWrapper implements LimiterWrapperProvider.
+func (f GetLimiterWrapperFunc) GetLimiterWrapper(key extensionlimiter.WeightKey, opts ...extensionlimiter.Option) (LimiterWrapper, error) {
+	if f == nil {
+		return PassThroughWrapper(), nil
+	}
+	return f(key, opts...)
+}
+
+var _ LimiterWrapperProvider = struct {
+	GetLimiterWrapperFunc
+	extensionlimiter.GetCheckerFunc
+}{}
+
+// LimiterWrapper is a general-purpose interface for limiter consumers
+// to limit resources with use of a callback.  This is the simplest
+// form of rate limiting interface from a callers perspective.  If the
+// caller is a pipeline component, consider using a consumer-oriented
+// limiterhelper (e.g., limiterhelper.NewLimitedLogs) to simplify
+// construction of this interface.
+//
+// A wrapped limiter is either a RateLimiter or ResourceLimiter
+// interface. LimiterWrappers can be constructed from either of the
+// underlying limiters and their corresponding providers. Usually
+// configmiddleware or limiterhelper is responsible for constructing
+// the correct wrapper from these two kinds of limiter; users will use
+// this interface consistently.
+type LimiterWrapper interface {
```

Comment:
Do you expect others to implement this interface? Otherwise we should have this as a struct.

Reply:
I see the point in the style you're after -- instead of mocking an interface, you should pass in an adapter and place a mock call inside your function.